src/base/stun.spec.ts

```diff
@@ -54,15 +94,23 @@ describe('test STUN', function () {
   })
 
   it('should perform a STUN request', async function () {
-    const multiAddrs = servers.map((server: ServerType) =>
-      Multiaddr.fromNodeAddress(nodeToMultiaddr(server.socket.address()), 'udp')
-    )
+    const multiAddrs = servers
+      .slice(1)
+      .map((server: ServerType) => Multiaddr.fromNodeAddress(nodeToMultiaddr(server.socket.address()), 'udp'))
 
     const result = await getExternalIp(multiAddrs, servers[0].socket)
 
+    // FIXME this fails when running behind a bidirectional NAT
```

Comment:
Worth to have a ticket and reference it from here

Reply:
changed

Reply:
Added a routine that adapts the test procedure if `bidirectional` NATs is detected